### PR TITLE
KATI-65_Refactor-API

### PR DESCRIPTION
**Pull request by vtnlinh95**

### CODE REVIEW

File: com.kms.cura.entity/src/com/kms/cura/entity/Entity.java [hunk 1]
```diff
@@ -24,4 +25,7 @@ public String getId() {
 	public void setId(String id) {
 		this.id = id;
 	}
+	public int getType() {
+		return -1;
```
Comment (hieumai) [inline com.kms.cura.entity/src/com/kms/cura/entity/Entity.java]: Don't return meaningless value like this. Should make this method abstract instead.

File: com.kms.cura_server/src/com/kms/cura_server/response/APIResponse.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+
+import com.google.gson.JsonObject;
+import com.kms.cura.entity.Entity;
+import com.kms.cura.entity.user.UserEntity;
+
+public abstract class APIResponse {
+
+	
+	public APIResponse(){
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/response/APIResponse.java]: Re-format this class

File: com.kms.cura_server/src/com/kms/cura_server/response/SpecialityAPIResponse.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.kms.cura.entity.Entity;
+import com.kms.cura.entity.SpecialityEntity;
+import com.kms.cura.entity.json.EntityToJsonConverter;
+import com.kms.cura.entity.user.UserEntity;
+
+public class SpecialityAPIResponse extends APIResponse{
+	
+	public SpecialityAPIResponse() {
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/response/SpecialityAPIResponse.java]: No need for constructor here, should remove it

File: com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.kms.cura.entity.Entity;
+import com.kms.cura.entity.json.EntityToJsonConverter;
+import com.kms.cura.entity.user.UserEntity;
+
+public class UserAPIResponse extends APIResponse{
+	
+	public UserAPIResponse() {
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java]: Should remove this constructor too

File: com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		// TODO Auto-generated constructor stub
+	}
+	@Override
+	public String successResponse(Entity entity) {
+		JsonElement JsonUser = EntityToJsonConverter.convertEntityToJson(entity);
+		JsonObject JsonUserConvert =(JsonObject) JsonUser;
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java]: Variable name should not start with uppercase.

File: com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import com.kms.cura.entity.user.UserEntity;
+
+public class UserAPIResponse extends APIResponse{
+	
+	public UserAPIResponse() {
+		// TODO Auto-generated constructor stub
+	}
+	@Override
+	public String successResponse(Entity entity) {
+		JsonElement JsonUser = EntityToJsonConverter.convertEntityToJson(entity);
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java]: Variable name should not start with upppercase.

File: com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		JsonUserConvert.addProperty(Entity.STATUS_KEY, true);
+		return JsonUserConvert.toString();
+	}
+	@Override
+	public String successResponse(List<? extends Entity> entity) {
+		return null;
+	}
+
+	public String successResponsewithType(Entity entity) {
+		JsonElement JsonUser = EntityToJsonConverter.convertEntityToJson(entity);
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java]: Variable name should not start with upppercase.

File: com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+		JsonObject JsonUserConvert =(JsonObject) JsonUser;
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/response/UserAPIResponse.java]: Variable name should not start with upppercase.

**Comment by vtnlinh95** [comment]
@hieumai  fixed

**Comment by vtnlinh95** [comment]
@hieumai fixed

**Comment by hieumai** [comment]
OK from me. @huuduy216  please verify this

**Comment by huuduy216** [comment]
okay from me as well

**Comment by hieumai** [comment]
@vtnlinh95 squash the commits so that I can merge

**Comment by vtnlinh95** [comment]
@hieumai  squashed

**Pull request merged**